src/dlgTriggerEditor.cpp

```diff
@@ -3469,18 +3469,22 @@ void dlgTriggerEditor::addVar(bool isFolder)
         mpVarsMainArea->comboBox_variable_value_type->setCurrentIndex(0);
     }
 
-    QStringList nameL;
-    nameL << QString();
-    QTreeWidgetItem* cItem = treeWidget_variables->currentItem();
     LuaInterface* lI = mpHost->getLuaInterface();
     VarUnit* vu = lI->getVarUnit();
-    TVar* cVar = vu->getWVar(cItem);
+
+    QStringList nameL;
+    nameL << QString(isFolder ? "New table name" : "New variable name");
+
     QTreeWidgetItem* pParent;
     QTreeWidgetItem* pNewItem;
-    if (cVar && cVar->getValueType() == LUA_TTABLE) {
-        pParent = cItem;
-    } else {
-        pParent = cItem->parent();
+    QTreeWidgetItem* cItem = treeWidget_variables->currentItem();
+    TVar* cVar = vu->getWVar(cItem);
```

Comment:
Doesn't this have to come *after* the next bit with the `if` - I mean, how does `getWVar(cItem)` behave if `cItem` is a `nullptr`?

Reply:
Oh, hang on, that doesn't fix the issue either - that too is using the `cITem` without handling the `nullptrtr` case (which is what will happen with no item being selected initially IIRC...

Reply:
all those problems apply to the old code as well. tbh, I'm not sure about all the cases (hence my posts about trying to remove this stuff from triggerEditor so we can actually test it better).

Reply:
there's no harm in embedding it in the if statement though. Looking at the code, cVar will return a nullptr is cItem is a nullptr

Reply:
I don't follow the rest of this about the nullptr not being handled -- this only sets the case for a parent point, which if it's null, the new item becomes a top level item. if (nullptr) resolves to false